spec/unit/plugins/vmware_spec.rb

```diff
@@ -0,0 +1,67 @@
+#
+# Author:: "Christopher M. Luciano" <cmlucian@us.ibm.com>
+# Copyright (C) 2015 IBM Corp.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+require 'spec_helper'
+
+describe Ohai::System, "plugin vmware" do
+  let (:plugin) do
+    plugin = get_plugin("vmware")
+  end
+
+  let(:path) { "/usr/bin/vmware-toolbox-cmd" }
+  let(:param) { nil }
+
+
+  context "vmware toolbox" do
+
+    def setup_stubs
+      allow(File).to receive(:exist?).and_return(true)
+      allow(plugin).to receive(:shell_out).with("/usr/bin/vmware-toolbox-cmd stat speed").and_return(mock_shell_out(0, "2000 MHz", nil))
+      allow(plugin).to receive(:shell_out).with("/usr/bin/vmware-toolbox-cmd stat hosttime").and_return(mock_shell_out(0, "04 Jun 2015 19:21:16", nil))
+      allow(plugin).to receive(:shell_out).with("/usr/bin/vmware-toolbox-cmd stat sessionid").and_return(mock_shell_out(0, "0x0000000000000000", nil))
+      allow(plugin).to receive(:shell_out).with("/usr/bin/vmware-toolbox-cmd stat balloon").and_return(mock_shell_out(0, "0 MB", nil))
+      allow(plugin).to receive(:shell_out).with("/usr/bin/vmware-toolbox-cmd stat swap").and_return(mock_shell_out(0, "0 MB", nil))
+      allow(plugin).to receive(:shell_out).with("/usr/bin/vmware-toolbox-cmd stat memlimit").and_return(mock_shell_out(0, "4000000000 MB", nil))
+      allow(plugin).to receive(:shell_out).with("/usr/bin/vmware-toolbox-cmd stat memres").and_return(mock_shell_out(0, "0 MB", nil))
+      allow(plugin).to receive(:shell_out).with("/usr/bin/vmware-toolbox-cmd stat cpures").and_return(mock_shell_out(0, "0 MHz", nil))
+      allow(plugin).to receive(:shell_out).with("/usr/bin/vmware-toolbox-cmd stat cpulimit").and_return(mock_shell_out(0, "4000000000 MB", nil))
+      allow(plugin).to receive(:shell_out).with("/usr/bin/vmware-toolbox-cmd upgrade status").and_return(mock_shell_out(0, "VMware Tools are up-to-date.", nil))
+      allow(plugin).to receive(:shell_out).with("/usr/bin/vmware-toolbox-cmd timesync status").and_return(mock_shell_out(0, "Disabled", nil))
+      plugin.run
```

Comment:
No, sorry. It would make more sense (just based off of the name of the method) to do

```ruby
before(:each) do
  setup_stubs
  plugin.run
end
```

Why do you call `get_vm_attributes` [here](https://github.com/chef/ohai/pull/551/files#diff-d9a909ded1b3f71262a5fed323232194R45)?